naptime/src/main/scala/org/coursera/naptime/Types.scala

```diff
@@ -91,34 +92,39 @@ object Types extends StrictLogging {
       case unknown: DataSchema =>
         throw new RuntimeException(s"Cannot compute asymmetric type for key type: $unknown")
     }
-    for ((name, graphQLRelation) <- fields.graphQLRelations) {
-      Option(mergedSchema.getField(name)) match {
-        case Some(field) =>
-          logger.warn(
-            s"Fields for resource $typeName tries to add graphql relation on field " +
-              s"'$name', but that field is already defined on the model")
-        case None =>
-          val errorMessageBuilder = new StringBuilder
-          val newField = graphQLRelation.toAnnotation.relationType match {
-            case FINDER | MULTI_GET =>
-              val newField = new RecordDataSchema.Field(new ArrayDataSchema(new StringDataSchema)) // TODO(bryan): fix type here
-              newField.setOptional(false)
```

Comment:
Fishing for an answer here: do we know why we set optional: false for finder/multigets (and now related resources) while it's true for single element finder /gets?

Reply:
We resolved in DMs, looks like it's because finder/multigets return arrays so we can always fall back to empty list. This is the case for related resources as well.